lib/at_cmd_parser/at_params.c

```diff
@@ -293,14 +326,44 @@ int at_params_string_get(const struct at_param_list *list, size_t index,
 
 	size_t param_len = at_param_size(param);
 
-	if (len < param_len) {
+	if (*len < param_len) {
 		return -ENOMEM;
 	}
 
 	memcpy(value, param->value.str_val, param_len);
-	return param_len;
+	*len = param_len;
+
+	return 0;
 }
 
+int at_params_array_get(const struct at_param_list *list, size_t index,
+			 u32_t *array, size_t *len)
+{
+	if (list == NULL || list->params == NULL || array == NULL) {
```

Comment:
*len output parameter should not be null.

Reply:
Ok